### DISCUSSION TITLE
Error when trying to apply terraform

### DISCUSSION THREAD

**Discussion by DavidBC71** [chadgeary/cloudblock#41] [Q&A] [answered]
![Screenshot 2022-02-14 141203](https://user-images.githubusercontent.com/98569606/153794302-6c34e16a-0097-4739-994b-0341d629ed6b.png)

Everything seemed to be progressing okay but I have this error when trying to apply terraform. Would be grateful for any assistance.

**Comment by chadgeary** [comment, marked as answer]
Terraform's error states the `source_id` is missing from the server (oci_core_instance). The source ID is defined in a terraform data lookup:

```
data "oci_core_image" "ph-image" {
  image_id                = var.oci_imageid
}
```

this data lookup requires variables `oci_imageid` which should be set in your `oci.tfvars` file:

https://github.com/chadgeary/cloudblock/blob/master/oci/oci.tfvars#L73

I talk about this variable at 6:57 in the oracle video: https://youtu.be/bVoO6XRNhJs?t=657

**Reply by DavidBC71** [reply]
Thanks. That seems to have worked and everything looks good. As far as I can tell this is all set up correctly. I'm not a techy guy but your instructions were very clear and even I was able to follow, in the most part. Do you have any recommendations for guides on how to set up the wireguard client on windows? Thanks again.
